.github/tools/cioutput.py

```
@@ -0,0 +1,666 @@
+	Creates log output compatible with GitHub Actions workflow commands
+	for creating annotations viewable in the GitHub UI.
+	"""
+
+	# Map between log levels and GitHub Actions annotation types
+	LEVEL_MAPPING = {
+		logging.DEBUG: "debug",
+		logging.INFO: "notice",
+		logging.WARNING: "warning",
+		logging.ERROR: "error",
+		logging.CRITICAL: "error"
+	}
```

Ruff (0.8.2)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)